src/core/backbone_router/bbr_manager.cpp

```diff
@@ -428,7 +428,7 @@ void Manager::SendDuaRegistrationResponse(const Coap::Message &      aMessage,
     SuccessOrExit(message->SetPayloadMarker());
 
     SuccessOrExit(Tlv::AppendUint8Tlv(*message, ThreadTlv::kStatus, aStatus));
-    SuccessOrExit(Tlv::AppendTlv(*message, ThreadTlv::kTarget, &aTarget, sizeof(aTarget)));
+    SuccessOrExit(Tlv::AppendTlv(*message, ThreadTlv::kTarget, aTarget));
```

Comment:
Instead of using template <typename ValueType> methods, is it better (and possible) to define methods like 
Tlv::AppendTlv\<ThreadTlv::kTarget\>(*message, aTarget) ? I think this can achieve better type checking. 

Thoughts?

Reply:
It is an interesting idea. I suggest we consider it as a potential future enhancement.

I am on the fence for this. On one hand, I like the type checking safety that this can add. On the other hand, I feel it may add a lot of complexity.

- Currently we have a common version of these method in `Tlv` base class (which is then shared/used by all)
- I think for this we would need a version like`AppendTlv<uint8_t TlvType, typename ValueType>` and then we need to add a template specialization for every simple TLV type (with its related `ValueType`)
- Also since the TLV integr type value are re-used by different TLV categories (e.g. TLV type value `0` is used as `Mle::Tlv::kSourceAddress = 0` and `MeshCop::Tlv::kChannel = 0` and `ThreadTlv::kTarget = 0`, etc.),  I don't think we can have a common template methods in `Tlv` base class and would need to have the template methods (re)defined by each TLV category (one set of methods in `Mle::Tlv`, one in `MeshCop::Tlv`, etc).

Reply:
I prefer defining template methods on each TLV category (e.g. Mle::TLv, ThreadTlv) so as to differentiate equal TLV types. 
But we can certainly leave it for further enhancement. 

Furthermore, I think this should be implemented based on `AppendTlv<uint8_t TlvType, typename ValueType>`.